app/src/main/resources/application-prod.yml

```diff
@@ -0,0 +1,11 @@
+server:
+  port: 9999
+
+invoicing-system:
+  database:
+    directory: db
+    type: file
+    id:
+      file: id.txt
+    invoices:
+      file: invoices.txt
```

Comment:
tutaj zrobiłbym ścieżke na odwrót, czyli zamiasta id.file i invoices.file to file.id i file.invoices, wtedy masz mniej rozgałęzień w drzewku i jest to też bardziej zrozumiałe przynajmniej dla mnie :)